tasks/main.yml

```diff
@@ -1,84 +1,14 @@
 ---
-- name: Checking Redis installation
-  shell: 'test -f {{ redis_path }}/bin/redis-cli && echo "yes" || echo "no"'
-  changed_when: False
-  register: installed_status
+- name: Ensure packages database is up to date
+  sudo: yes
+  apt: update_cache=yes
 
-- name: Get Redis installed version
-  shell: "{{ redis_path }}/bin/redis-cli -v | awk {'print $2'}"
-  when: installed_status.stdout == "yes"
-  register: installed_version
-
-- name: Set installed variable
-  set_fact: is_installed='{{ installed_status.stdout == "yes" and installed_version.stdout == redis_version }}'
-
-- name: Install dependencies
-  apt: pkg={{ item }} state=latest
-  with_items:
-    - make
-
-- name: Download
-  get_url: >
-    url=http://download.redis.io/releases/redis-{{ redis_version }}.tar.gz
-    dest=/tmp/redis-{{ redis_version }}.tar.gz
-  when: not is_installed
-
-- name: Extract
-  unarchive: >
-    src=/tmp/redis-{{ redis_version }}.tar.gz
-    dest=/tmp
-    copy=no
-  when: not is_installed
-
-- name: Compile
-  command: make -j5
-           chdir=/tmp/redis-{{ redis_version }}
-           creates=/tmp/redis-{{ redis_version }}/src/redis-server
-  when: not is_installed
-
-- name: Create directory
-  file: path={{ item }} state=directory
-  with_items:
-    - "{{ redis_path }}-{{ redis_version }}"
-    - "{{ redis_conf_path }}"
-    - "{{ redis_pid_path }}"
-    - "{{ redis_db_path }}/{{ redis.port }}"
-
-- name: Install
-  command: make PREFIX={{ redis_path }}-{{ redis_version }} install
-           chdir=/tmp/redis-{{ redis_version }}
-           creates={{ redis_path }}-{{ redis_version }}/bin/redis-server
-  when: not is_installed
-
-- name: Link Redis folder
-  file: src={{ redis_path }}-{{ redis_version }} dest={{ redis_path }} state=link
-
-- name: Create init script
-  template: src=init.tpl dest=/etc/init.d/redis mode=0755
-
-- name: Set to start boot
-  service: name=redis enabled=yes
+- name: Install Redis
+  apt: name=redis-server state=installed
 
 - name: Create config file
-  template: src=conf.tpl dest={{ redis_conf_path }}/{{ redis.port }}.conf
+  template: src=conf.tpl dest={{ redis_conf_path }}/{{ redis_port }}.conf
```

Comment:
redis_port is better because allows you to change only one var from defaults instead of all the array.

Last time I've checked was te recommended way.


Anyway I was asking for retro compat.
We can set an issue on phansible to normalize this later.